pkg/alertmanager/operator.go

```diff
@@ -473,6 +482,22 @@ func AlertmanagerStatus(kclient *kubernetes.Clientset, a *v1alpha1.Alertmanager)
 	return res, oldPods, nil
 }
 
+func needsUpdate(pod *v1.Pod, tmpl v1.PodTemplateSpec) bool {
+	c1 := pod.Spec.Containers[0]
+	c2 := tmpl.Spec.Containers[0]
+
+	if c1.Image != c2.Image {
+		return true
+	}
+
+	// TODO: Prometheus TPR checks on Args which seems to be the sane thing to do.
```

Comment:
just fyi this will cause a redeploy on Prometheus Operator upgrades, but I'm fine with switching form command to args .. not even sure why we didn't do that in the first place